blueocean-core-js/src/js/paths/rest.js

```diff
@@ -23,7 +27,7 @@ export default {
 
     run({ organization, pipeline, branch, runId }) {
         if (branch) {
-            return `${this.pipeline(organization, pipeline)}branches/${branch}/runs/${runId}/`;
+            return `${this.pipeline(organization, pipeline)}branches/${this.encodeSlashes(branch)}/runs/${runId}/`;
```

Comment:
also this encode function must be unit tested as well, and not be specific to just slashes necessarily. What other branch name things may need to be encoded, what was in use before etc.